lib/cmd/common/admin-devices.js

```diff
@@ -13,108 +12,97 @@ devices.usage = "\nfhc devices list"
 
 var fhreq = require("../../utils/request");
 var common = require("../../common");
+var _ = require('underscore');
+
+var actions = {
+  list: list,
+  read: read,
+  update: update,
+  enable: enabledDevice,
+  purgedata: purgedataDevice,
+  listusers: listusers,
+  listapps: listapps
+};
 
 // Main devices entry point
-function devices (argv, cb) {
+function devices(argv, cb) {
+  /*eslint no-else-return:0 */
   var args = argv._;
-  if (args.length === 0){
+  if (args.length === 0) {
     return list(cb);
   }
 
-  var action = args[0];
-  if (action === 'list') {
-    return list(cb);
-  }else if (action === 'read'){
-    if (args.length !== 2) return cb("Invalid arguments for 'read':" + devices.usage);
-    return read(args[1], cb);
-  }else if (action === 'update'){
-    if (args.length !== 3) return cb("Invalid arguments for 'update':" + devices.usage);
-    return update(args[1], args[2], cb);
-  }else if (action === 'enable'){
-    if (args.length !== 3) return cb("Invalid arguments for 'enable':" + devices.usage);
-    return enabledDevice(args[1], args[2], cb);
-  }else if (action === 'purgedata'){
-    if (args.length !== 3) return cb("Invalid arguments for 'purgedata':" + devices.usage);
-    return purgedataDevice(args[1], args[2], cb);
-  }else if (action === 'listusers'){
-    if (args.length !== 2) return cb("Invalid arguments for 'listusers':" + devices.usage);
-    return listusers(args[1], cb);
-  }else if (action === 'listapps'){
-    if (args.length !== 2) return cb("Invalid arguments for 'listapps':" + devices.usage);
-    return listapps(args[1], cb);
-  }else if (args.length === 1){
-    var deviceId = args[0];
-    return read(deviceId, cb);
-  }else{
+  var action = args.shift();
+  var func = actions[action];
+  if (!func) {
+    // command not found, try `read({deviceId})`
+    if (args.length === 1) {
+      var deviceId = args[0];
+      return read(deviceId, cb);
+    }
     return cb(devices.usage);
   }
+  if (args.length !== func.length + 1) {
+    return cb("Invalid arguments for '" + action + "':" + devices.usage);
+  }
+  return func.command.apply(undefined, args.slice(func.length).concat(cb));
 }
```

Comment:
Thanks @wtrocki, I've taken a look at the structure for the newer commands and I agree that these should be refactored to that. This is outside of the scope of this issue, so I'll revert this and we'll eventually get to it. (along with better testing, linting, etc.)